pgtest.go

```diff
@@ -142,9 +188,28 @@ func start(config *PGConfig) (*PG, error) {
 	// Initialize PostgreSQL data directory
 	_, err = os.Stat(filepath.Join(dataDir, "postgresql.conf"))
 	if os.IsNotExist(err) {
-		init := prepareCommand(isRoot, filepath.Join(binPath, "initdb"),
+		args := []string{
 			"-D", dataDir,
 			"--no-sync",
+		}
+
+		// setup password if specified
+		if config.Password != "" {
+			pwdFile, cleanupPwdFile, err := createTempPasswordFile(config.Password, pgUID, pgGID)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create password file: %w", err)
+			}
+			// for more info on how this is being done
+			// https://www.postgresql.org/docs/current/app-initdb.html#APP-INITDB-OPTION-AUTH
+			// https://www.postgresql.org/docs/current/app-initdb.html#APP-INITDB-OPTION-PWFILE
+			args = append(args, "-A", "scram-sha-256", fmt.Sprintf("--pwfile=%s", pwdFile))
+			// remove the password file, when we return from start()
+			defer cleanupPwdFile()
+		}
```

Comment:
When specifying a password for the database, we need to write the password in a file and pass the path of the file to `initdb` using the flag `--pwdfile=<path>` along with the option `-A scram-sha-256`.

[read more about -A option](https://www.postgresql.org/docs/current/app-initdb.html#APP-INITDB-OPTION-AUTH)
[read more about --pwdfile flag](https://www.postgresql.org/docs/current/app-initdb.html#APP-INITDB-OPTION-PWFILE)


So we create a temp file, put the password in it, then we pass that file to `initdb` and we remove that file before the main `start()` function returns.